src/makeFlat.js

```diff
@@ -0,0 +1,42 @@
+import _isArrayLike from './isArrayLike';
+
+/**
+ * `makeFlat` is a helper function that returns a one-level or fully recursive
+ * function based on the flag passed in.
+ *
+ * @func makeFlat
+ * @memberOf RA
+ *
+ * @category List
+ * @param {!bool} = should recursively flatten
+ * @param {!Array} = the nested list to be flattened
+ * @return {!Array} = the flattened list
+ * @sig Bool -> List -> List
+ *
+ */
+export default function makeFlat(recursive) {
```

Comment:
Let's make the function declaration more modern.
```js
const makeFlat = recursive => list  => {
  ...body of the function..
};
```

Reply:
how do you suppose calling the recursion on an anonymous function?

Reply:
Right, you're absolutely correct. Never mind my comment.